plugin/src/main/java/com/denizenscript/denizen/events/entity/EntityEntersBlockScriptEvent.java

```diff
@@ -0,0 +1,91 @@
+package com.denizenscript.denizen.events.entity;
+
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizen.objects.EntityTag;
+import com.denizenscript.denizen.objects.LocationTag;
+import com.denizenscript.denizen.objects.MaterialTag;
+import com.denizenscript.denizen.utilities.implementation.BukkitScriptEntryData;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.scripts.ScriptEntryData;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.EntityEnterBlockEvent;
+
+public class EntityEntersBlockScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // <entity> enters <block>
+    //
+    // @Group Entity
+    //
+    // @Location true
+    //
+    // @Cancellable true
+    //
+    // @Triggers when an entity enters, and is stored in a block (EG a bee enters a bee nest)
+    // Does not fire when a silverfish "enters" a stone block. Prefer <@link event <entity> changes <block> (into <block>)> for that.
+    //
+    // @Context
+    // <context.entity> returns the EntityTag.
+    // <context.location> returns the LocationTag of the block entered by the entity.
+    // <context.material> returns the MaterialTag of the block entered by the entity.
+    //
+    // @Player when the entity that entered the block is a player

```

Comment:
Can a player enter a block? Per javadocs it says `This event is called for bees entering a bee hive.`

Reply:
That was mentioned on Discord thread; wasn't sure if I should have these or not

It is definitely not possible in Vanilla but I wasn't sure if there's potentially some weird way a Player or NPC could do this via plugins that would need this

Reply:
Given that it's not expected to be possible in any normal circumstance, the docs should exclude it